packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2910,15 +2911,38 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this.baseSnapshotBlobs = await SerializedSnapshotStorage.serializeTree(this.context.baseSnapshot, this.storage);
     }
 
+    private validatePendingMessages(pendingMessages: IPendingLocalState | undefined): void {
+        if (pendingMessages === undefined) {
+            return;
+        }
+        const firstMessage = pendingMessages.pendingStates[0] as IPendingMessage;
+        const isFirstMessageBatchFirst = firstMessage.opMetadata?.batch;
+        const lastMessage = pendingMessages.pendingStates[pendingMessages.pendingStates.length - 1].type;
+        if (isFirstMessageBatchFirst == null) {
+            throw new UsageError("First message must have batch metadata");
+        }
+        if (!isFirstMessageBatchFirst) {
+            throw new UsageError(`First message must be first in batch, opMetadata.batch: ${isFirstMessageBatchFirst}`);
+        }
+        if (lastMessage !== "flush") {
+            throw new UsageError(`"Last message in batch must be flush, type: ${lastMessage}"`);
+        }
```

Comment:
you should also log when it happens.

Also, you can throw (and log) one single error for this. The client shouldn't really care about the batching insider information. I mean I'm calling this and receiving `Last message in batch must be flush`. What do I do with that information? What's a flush? This information is more relevant to us than to the customer, so we should log it.

One approach would be something like

```
if (isFirstMessageBatchFirst == null || !isFirstMessageBatchFirst || lastMessage !== "flush") {
    const error = new UsageError("Cannot call validatePendingMessages while the runtime is processing a message batch");
    logger.sendErrorEvent({
          eventName: "getPendingLocalStateDuringBatch",
          hasBatchStart: isFirstMessageBatchFirst != null, // or whatever field name is better
          firstMessageStartOfBatch: isFirstMessageBatchFirst, // or whatever field name is better
          lastMessageFlush: lastMessage === "flush", // or whatever field name is better
      },
      error);
      throw error;
}
```

this way, if this happens, the fields in the event in Kusto can give us some insight into why this failed but won't confuse the caller with batching details.